lib/codegen/generator-soap.js

```diff
@@ -0,0 +1,429 @@
+      if (outputBodyDescriptor.qname) {
+        modelName = outputBodyDescriptor.qname.name;
+      }            else { // this may be a case of default namespace. Fall back on using input message name
+        modelName = op.output.message.$name;
+      }
+      buildElementProperties(models, modelName, {}, outputBodyDescriptor, schemaTypes);
+    }
+  }
+
+  return models;
+};
+
+// builds properties for input/output params. It recursively builds properties for elements defined in the schema
+function buildElementProperties(models, modelName, propertyList, outerElem, schemaTypes) {
+  models[modelName] = {
+    'name': modelName,
+    'properties': propertyList,
+  };
+  var elements = outerElem.elements;
+  for (let i in elements) {
+    var element = elements[i];
+    var type;
+
+    var propertyData;
+    var modName;
+    if (!element.type && !element.isSimple) {
+      modName = element.qname.name;
+      buildElementProperties(models, modName, {}, element, schemaTypes);
+            // check refOriginal since 'type' is retuned as 'undefined' if it's referenced in a nested imported schema
+      if (element.refOriginal) {
+        propertyData = {
+          type: element.refOriginal.qname.name,
+        };
+        propertyList[element.qname.name] = propertyData;
+      }
+      continue;
+    }
+        // if simpleType, check in simpleTypes list. For e.g Enum could be of simpleType, but not xs: simple type
+    if (element.type) {
+      if (schemaTypes.simpleTypes && schemaTypes.simpleTypes[element.type.name]) {
+        type = schemaTypes.simpleTypes[element.type.name].children[0].base.$name;
+      } else {
+                // could be a complextype or xs: simple type
+        type = element.type.name;
+      }
+    }
+    propertyData = {
+      type: type,
+    };
+    propertyList[element.qname.name] = propertyData;
+
+    if (!element.isSimple) {
+      modName = element.type.name;
+      buildElementProperties(models, modName, {}, element, schemaTypes);
+    }
+  }
+  return models;
+}
+
+// get operations in a specific format needed for code generation in model.ejs template
+function getOperations(spec) {
+  var models = spec.definitions;
+
+  var operations = {};
+
+  for (var path in spec.paths) {
+    if (path.indexOf('x-') === 0) continue;
+    var ops = spec.paths[path];
+        /* eslint-disable one-var */
+    for (var verb in ops) {
+      var op = ops[verb];
+
+      if (!op.parameters) {
+        op.parameters = [];
+      }
+
+      op.models = models;
+      op.verb = verb.toLowerCase();
+      op.path = path.replace(/{(([^{}])+)}/g, ':$1');
+
+            // operationId is optional
+      if (!op.operationId) {
+                // Derive the operationId from verb & path
+        op.operationId = op.verb.toLowerCase() + '_' + op.path;
+      }
+
+            // Camelize the operation id
+      op.operationId = op.operationId.replace(/{(([^{}])+)}/g, '_$1');
+      op.operationId = _s.camelize(
```

Comment:
It looks like this one call is the only reason `underscore.string` is used at all. Does it do something that `_.camelCase` from lodash doesn't do?